### PR TITLE
Expose recently introduced map_array function in skimage.util

### PR DESCRIPTION
**Pull request by emmanuelle**
Since this function can be useful for other tasks. I also wondered whether we should expose `ArrayMap` or not (at the moment it is not and for skimage developers it's easy to import it from its location). It's easy to change of course.

### CODE REVIEW

File: skimage/util/__init__.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 from .unique import unique_rows
 from ._invert import invert
 from ._montage import montage
-
+from ._map_array import map_array
```
Comment (jni) [inline skimage/util/__init__.py:17]:
```suggestion
from ._map_array import map_array

```
PEP8

File: skimage/segmentation/_join.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import numpy as np
 from .._shared.utils import deprecated
-from ._remap import _map_array
+from ..util import map_array
+from ..util._map_array import ArrayMap
```
Comment (jni) [inline skimage/segmentation/_join.py:4]:
```suggestion
from ..util._map_array import map_array, ArrayMap
```

I think it's better to use direct imports in our own code...

**Comment by jni** [comment]
To clarify what @emmanuelle meant by "other tasks": this would close #1396. It needs a gallery example but it's probably not worth holding up release for it.

File: skimage/segmentation/_join.py (hunk 2 above)
Comment (emmanuelle) [reply skimage/segmentation/_join.py:4]: oh interesting I didn't know.

**Comment by rfezzani** [comment]
I am thinking about submitting a PR to create a `skimage.labelling` submodule where `skimage.morphology.label`/`skimage.measure.label`, `skimage.segmentation.relabel_sequential`, `skimage.color.colorlabel`... can take place. `MapArray` was on list too but `skimage.util` is a good place ;-).
I am waiting for the CI to be green before approving...

**Comment by emmanuelle** [comment]
It is green now :-).

**Comment by rfezzani** [comment]
Merged ;) Thank you @emmanuelle

**Comment by jni** [comment]
@rfezzani I like your new module idea. `labeltools`? Anyway, I think the best time for this transition would be the 1.0 release. =)

**Comment by rfezzani** [comment]
Yes @jni, there is no hurry ;) `labeltools` is good for me :+1:

File: skimage/segmentation/_join.py (hunk 2 above)
Comment (jni) [reply skimage/segmentation/_join.py]: It's a new in-trial policy from #3397 =)

**Comment by JDWarner** [comment]
Continuing the tangent, our top-level submodule naming convention is direct, not plural, and tends not to include portmanteaus.

Refactoring the label infrastructure makes some sense to me.  The best submodule name would be `skimage.label` IMHO, but `label` already being a function name clouds the picture a bit (`from skimage.label import label` feels somewhat cumbersome).

**Pull request merged**